src/actors/symbolication.rs

```diff
@@ -152,6 +172,203 @@ impl SymbolicationActor {
             result,
         )
     }
+
+    fn do_process_minidump(
+        &mut self,
+        request: ProcessMinidump,
+    ) -> impl Future<Item = SymbolicationResponse, Error = SymbolicationError> {
+        let ProcessMinidump {
+            file,
+            scope,
+            sources,
+        } = request;
+
+        let byteview = tryf!(ByteView::read(file).map_err(|_| SymbolicationError::Minidump));
+
+        let object_infos = self.threadpool.spawn_handle(future::lazy(move || {
+            log::info!("Minidump size: {}", byteview.len());
+            metric!(time_raw("minidump.upload.size") = byteview.len() as u64);
+            // XXX: Use ByteView::from_file
+            let state = ProcessState::from_minidump(&byteview, None)
+                .map_err(|_| SymbolicationError::Minidump)?;
+
+            let os_name = state.system_info().os_name();
+
+            Ok((
+                byteview,
+                state
+                    .referenced_modules()
+                    .into_iter()
+                    .filter_map(|code_module| {
+                        Some((
+                            code_module.id()?,
+                            object_info_from_minidump_module(&os_name, code_module),
+                        ))
+                    })
+                    .collect::<Vec<_>>(),
+            ))
+        }));
+
+        let cficaches = &self.cficaches;
+
+        let cfi_requests = object_infos.and_then(clone!(cficaches, scope, sources, |(
+            byteview,
+            object_infos,
+        )| {
+            join_all(
+                object_infos
+                    .into_iter()
+                    .map(move |(code_module_id, object_info)| {
+                        cficaches
+                            .send(FetchCfiCache {
+                                object_type: object_info.ty.clone(),
+                                identifier: object_id_from_object_info(&object_info),
+                                sources: sources.clone(),
+                                scope: scope.clone(),
+                            })
+                            .map_err(|_| SymbolicationError::Mailbox)
+                            .map(move |result| (object_info, code_module_id, result))
+                    }),
+            )
+            .map(move |cfi_requests| (byteview, cfi_requests))
+        }));
+
+        let threadpool = &self.threadpool;
+
+        let symbolication_request =
+            cfi_requests.and_then(clone!(threadpool, scope, sources, |(
+                byteview,
+                cfi_requests,
+            )| {
+                threadpool.spawn_handle(future::lazy(move || {
+                    let mut frame_info_map = FrameInfoMap::new();
+                    let mut modules = vec![];
+
+                    for (object_info, code_module_id, result) in &cfi_requests {
+                        // XXX: We should actually build a list of FetchedDebugFile instead of
```

Comment:
TODO

Reply:
I concluded we need a separate field in `ObjectInfo` for representing the status of fetching the CFI file. This might yield entirely different errors than fetching the debug file during symbolication.

Reply:
deferring